present/slideshow.py

```diff
@@ -23,11 +28,33 @@
 )
 
 
+def render_code_block(screen, block, row):
+    # Divide the width by 3
+    column = int(screen.dimensions[1] / 3)
+    lexer = get_lexer_by_name(block.lang())
+
+    for cur_row, line in enumerate(block.padded_lines(), start=row):
+        coded_text = highlight(line, lexer, Terminal256Formatter())
+        text = ColouredText(
+            coded_text,
+            AnsiTerminalParser(),
+        )
+        screen.paint(
+            text,
+            column,
+            cur_row,
+            colour_map=text.colour_map,
+        )
+
+
 class Slide(Scene):
-    def __init__(self, show, effects, fg_color, bg_color):
+    def __init__(self, show, slide):
         self.show = show
-        self.fg_color = fg_color
-        self.bg_color = bg_color
+        self.fg_color = slide.fg_color
+        self.bg_color = slide.bg_color
+        self.code_blocks = slide.code_blocks
+        effects = show.build_scene(slide)
```

Comment:
Since a code block is no longer considered an effect, this refactoring is an attempt to remove some mystery around effects, `Slideshow`, and the two `Slide` classes.

I renamed `get_effects` to `build_scene` since I'm using it to also add to a slide's `code_blocks` list.